ts/main.ts

```diff
@@ -63,8 +63,145 @@ $entryForm.addEventListener('submit', function (event: Event) {
+
+  const $colHalfDiv1 = document.createElement('div');
+  $colHalfDiv1.setAttribute('class', 'column-half');
+  const $entryImg = document.createElement('img');
+  $entryImg.setAttribute('class', 'list-img');
+  $entryImg.setAttribute('src', entry.photoURL);
+
+  const $colHalfDiv2 = document.createElement('div');
+  $colHalfDiv2.setAttribute('class', 'column-half');
+  const $entryTitle = document.createElement('h3');
+  $entryTitle.textContent = entry.title;
+  const $entryNotes = document.createElement('p');
+  $entryNotes.textContent = entry.notes;
+
+  $colHalfDiv1.appendChild($entryImg);
+  $colHalfDiv2.appendChild($entryTitle);
+  $colHalfDiv2.appendChild($entryNotes);
+  $rowDiv.appendChild($colHalfDiv1);
+  $rowDiv.appendChild($colHalfDiv2);
+  $entryLi.appendChild($rowDiv);
+
+  return $entryLi;
+}
+
+const $entriesList = document.querySelector(
+  '.entries-list',
+) as HTMLUListElement;
+if (!$entriesList) throw new Error('.entries-list query failed!');
+
+document.addEventListener('DOMContentLoaded', function () {
+  if (data.view === 'entries') {
+    viewSwap('entries');
+  } else if (data.view === 'entry-form') {
+    viewSwap('entry-form');
+  }
+
+  for (let i = 0; i < data.entries.length; i++) {
+    const entry = renderEntry(data.entries[i]);
+    $entriesList.prepend(entry);
+  }
+  if (
+    data.entries.length === 0 &&
+    $noEntriesMessage.classList.contains('hidden')
+  ) {
+    toggleNoEntries();
+  } else if (
+    data.entries.length > 0 &&
+    !$noEntriesMessage.classList.contains('hidden')
+  ) {
+    toggleNoEntries();
+  }
+});
+
+const $noEntriesMessage = document.querySelector(
+  '.no-entries-message',
+) as HTMLHeadingElement;
+if (!$noEntriesMessage) throw new Error('.no-entries-message query failed!');
+
+function toggleNoEntries(): void {
+  if ($noEntriesMessage.classList.contains('hidden')) {
+    $noEntriesMessage.classList.remove('hidden');
+  } else {
+    $noEntriesMessage.classList.add('hidden');
+  }
+}
+
+const $entryFormView = document.querySelector(
+  '.entry-form-view',
+) as HTMLDivElement;
+if (!$entryFormView) throw new Error('.entry-form-view query failed!');
+const $entriesView = document.querySelector('.entries-view') as HTMLDivElement;
+if (!$entriesView) throw new Error('entries-view query failed!');
+
+function viewSwap(viewName: string): void {
+  if (viewName === 'entries') {
+    $entryFormView.classList.add('hidden');
+    $entriesView.classList.remove('hidden');
+    data.view = 'entries';
+  }
+  if (viewName === 'entry-form') {
+    $entriesView.classList.add('hidden');
+    $entryFormView.classList.remove('hidden');
+    data.view = 'entry-form';
+  }
+}
+
+const $viewSwapAnchor = document.querySelector(
+  '.view-swap-anchor',
+) as HTMLAnchorElement;
+if (!$viewSwapAnchor) throw new Error('.view-swap-anchor query failed!');
+$viewSwapAnchor.addEventListener('click', function (event: Event): void {
+  event.preventDefault();
+  viewSwap('entries');
```

Comment:
Do you need to prevent the default functionality of this here? It's not a form submission but just an anchor.